src/pages/iou/request/step/IOURequestStepAmount.tsx

```diff
@@ -290,9 +281,13 @@ function IOURequestStepAmount({
             return;
         }
 
-        const taxAmount = getTaxAmount(transaction, policy, newAmount);
+        // If currency has changed, then we get the default tax rate based on currency, otherwise we use the current tax rate selected in transaction, if we have it.
+        const transactionTaxCode = transaction?.taxCode ?? '';
+        const taxCode = currency !== transactionCurrency ? TransactionUtils.getDefaultTaxCode(policy, transaction, currency) ?? '' : transactionTaxCode;
```

Comment:
Since `transactionTaxCode` may return empty, can we fallback to `defaultTaxCode` as `getDefaultTaxCode()` is always guaranteed to return a valid value? 
Something like this:

```
const defaultTaxCode = TransactionUtils.getDefaultTaxCode(policy, transaction, currency);
const taxCode = (currency !== transactionCurrency ? defaultTaxCode : transactionTaxCode) ?? defaultTaxCode;
```

This way, we can avoid `taxCode` being empty. This also aligns with the existing implementation i.e. `transactionTaxCode ?? defaultTaxCode` where we fallback to the default one if `transactionTaxCode` is empty. This will also help us keep any possible regressions at bay.

Reply:
Good point. Updated!